### PR TITLE
fix statistics charts render

### LINKED ISSUE #10: Fix statistics after deleting a round
### Environment
Browsers: Chrome 
OS: Mac OS

### Steps to reproduce
Click on edit Round button and delete it.

image

### Expected Behavior
Statistics is visible and updated 

### Actual Behavior
Statistics tab is empty:

image

### PR DESCRIPTION
**Pull request by YasiOnFire**
fixes #10 

Changed chart update logic, fixing disappearing statistics tab.

### CODE REVIEW

File: client/src/components/ChartBars.vue [hunk 1]
```diff
@@ -12,29 +12,47 @@ export default {
...
   },
   mounted() {
     this.setChart();
   },
   methods: {
+    updateChart () {
+      if (this.$data._chart) {
+        this.$data._chart.data = this.chartdata;
+        this.$data._chart.update();
+      }
```
Comment (DavidGolodetsky) [inline client/src/components/ChartBars.vue:51]: you don't need $data. You can go for this._chart directly

File: client/src/components/ChartBars.vue [hunk 2]
```diff
@@ -12,29 +12,47 @@ export default {
...
+    },
+  },
   watch: {
-    rounds(value) {
-      value ? this.setChart() : this._chart.destroy();
+    rounds() {
+      this.updateChart()
+    },
+    tab(val) {
+      if (val === 'tab-1') this.updateChart()
```
Comment (DavidGolodetsky) [inline client/src/components/ChartBars.vue:40]: Can we omit this strong coupling with tabs? This component could be used as a standalone in the future.

**Review by DavidGolodetsky — CHANGES_REQUESTED** [review]

File: client/src/components/ChartBars.vue (hunk 1 above)
Comment (YasiOnFire) [reply client/src/components/ChartBars.vue:51]: This does not work for me without the `$data` (and I'm certain it should not). `this._chart` is `undefined`. 
Docs: https://vue-chartjs.org/api/#chart-js-object
Source: https://github.com/apertureless/vue-chartjs/blob/develop/src/BaseCharts.js

File: client/src/components/ChartBars.vue (hunk 2 above)
Comment (YasiOnFire) [reply client/src/components/ChartBars.vue:40]: Jup, you're right, I will remove it. It was redundant anyways, fix works without it.

File: client/src/components/ChartBars.vue (hunk 1 above)
Comment (DavidGolodetsky) [reply client/src/components/ChartBars.vue:48]: fair enough:)

File: client/src/views/TeamDetails.vue [hunk 3]
```diff
@@ -31,7 +31,7 @@
           :key="i"
           :value="`tab-${i}`"
         >
-          <component :is="component" :team="team" :rounds="rounds" />
+          <component :is="component" :team="team" :rounds="rounds" :tab="tab" />
```
Comment (DavidGolodetsky) [inline client/src/views/TeamDetails.vue:34]: Could you remove this as well?

File: client/src/components/ChartBars.vue [hunk 4]
```diff
@@ -12,29 +12,44 @@ export default {
     rounds: {
       type: Array,
       required: true
+    },
+    tab: {
+      type: String,
+      required: true
```
Comment (DavidGolodetsky) [inline client/src/components/ChartBars.vue:18]: Could you remove this as well?

**Review by DavidGolodetsky — CHANGES_REQUESTED** [review]

**Comment by DavidGolodetsky** [comment]
Nice job! Thank you @YasiOnFire

**Pull request merged**